src/rust/catmem/mod.rs

```diff
@@ -112,6 +117,22 @@ impl CatmemLibOS {
         Ok(())
     }
 
+    /// Half-closes a memory queue.
```

Comment:
It would be good to document what half-close means in this context. I think it drops the reference to the underlying pipe but doesn't close it. If that is the case, then we could rename it something more meaningful like free_qd()? I'm not sure that is the best name, but something like that.

Reply:
I added a comment to give more details on this operation.

On the function name, we can have a discussion on it, but I don't believe that `free_qd()`, because this is actually the equivalent of disallowing further receptions on this pipe, like `shutdown(fd, SHUT_RD)`.

Reply:
Should this be close_endpoint()? Semantically, close and shutdown have the same meaning, so we should try to come up with a more descriptive name. It seems like it closes the local end of the pipe, so I think close_endpoint() could make sense or close_local().

Reply:
OK, now that I have seen the documentation for shutdown on sockets, I agree the function should be called shutdown.